board/safety/safety_subaru.h

```diff
@@ -165,7 +187,27 @@ static int subaru_tx_hook(CANPacket_t *to_send) {
     violation |= steer_torque_cmd_checks(desired_torque, -1, limits);
   }
 
-  if (violation){
+  if (subaru_longitudinal) {
+    // check es_brake brake_pressure limits
+    if (addr == MSG_SUBARU_ES_Brake) {
+      int es_brake_pressure = ((GET_BYTES(to_send, 0, 4) >> 16) & 0xFFFFU);
+      violation |= longitudinal_brake_checks(es_brake_pressure, SUBARU_LONG_LIMITS);
+    }
+
+    // check es_distance cruise_throttle limits
+    if (addr == MSG_SUBARU_ES_Distance) {
+      int cruise_throttle = ((GET_BYTES(to_send, 0, 4) >> 16) & 0xFFFU);
+      violation |= longitudinal_gas_checks(cruise_throttle, SUBARU_LONG_LIMITS);
+    }
+
+    // check es_status cruise_rpm limits
+    if (addr == MSG_SUBARU_ES_Status) {
+      int cruise_rpm = ((GET_BYTES(to_send, 0, 4) >> 16) & 0xFFFU);
```

Comment:
simplify this, it's getting two extra bytes then immediately shifting them out

Reply:
done